src/v/redpanda/admin_server.cc

```diff
@@ -525,3 +530,44 @@ void admin_server::register_status_routes() {
           return ss::make_ready_future<ss::json::json_return_type>(status_map);
       });
 }
+
+std::vector<model::node_id> parse_node_ids(const ss::sstring& params) {
+    std::vector<model::node_id> ids;
+    std::string token;
+    std::istringstream tokenStream(params);
+    while (std::getline(tokenStream, token, ',')) {
+        ids.emplace_back(boost::lexical_cast<int32_t>(token));
+    }
+    return ids;
+}
+
+void admin_server::register_node_routes() {
+    ss::httpd::node_json::decommission_nodes.set(
+      _server._routes, [this](std::unique_ptr<ss::httpd::request> req) {
+          auto ids_str = req->query_parameters.find("node_id")->second;
```

Comment:
What if the find returns collection.end()? i.e. element 'node_id' doesn't exist in the query params